.ci/docker.run

```diff
@@ -77,6 +77,55 @@ make -j check
 
 popd
 
+#
+# Backwards compatibility test for ensuring we do not break the tool options
+# determined in 4.X release.
+#
+if [[ "${TPM2_TSS_VERSION}" == "master" && "$CC" != clang* ]];then
+  echo "tpm2-tools v4.X compatibility test follows"
+  git fetch origin refs/tags/4.3.0:refs/tags/4.3.0
+  git checkout 4.3.0 test/integration/tests
+  git clean -fdx tests/integration/fapi
+  #
+  # RSA OAEP decryption was enabled as a feature after 4.X and so testing it
+  # would signal a false compatibility test failure.
+  #
+  git checkout HEAD test/integration/tests/rsadecrypt.sh
+  #
+  # tpm2_getekcertificate is known to break backwards compatibility
+  #
+  git checkout HEAD test/integration/tests/getekcertificate.sh
+  #
+  # symlink is an irrelevant test for 4.X branch
+  #
+  rm test/integration/tests/symlink.sh
+
+  #
+  # Beyond 4.X release, the tpm2-tools were combined into a single
+  # busybox style binary "tpm2". The following makes adjustments
+  # to the tool-name which essentially invokes the same tools.
+  #
+  for f in `find test/integration/tests -iname '*.sh'`
+  do
+      for i in `find tools -iname 'tpm2*.c'`
+      do
+          test=$(basename $i .c)
+          replace=$(basename $i .c | sed  's/tpm2_//g')
+          sed -i "s/$test/tpm2 $replace/g" $f
+      done
+  done
+
+  ./bootstrap
+
+  mkdir compatibility_testbuild
+  pushd compatibility_testbuild
+  ../configure --enable-unit --disable-fapi --disable-hardening --with-tpmsim=tpm_server
```

Comment:
configure.ac could be missing a commit to enable "--with-tpmsim" option. However, we don't need this option because this 4.X branch with used tpm_server anyway.